packages/web/src/SessionBasedSampler.ts

```diff
@@ -72,20 +71,22 @@ export class SessionBasedSampler implements Sampler {
 		traceId: string,
 		spanName: string,
 		spanKind: SpanKind,
-		attributes: SpanAttributes,
+		attributes: Attributes,
 		links: Link[],
 	): SamplingResult {
 		// Implementation based on @opentelemetry/core TraceIdRatioBasedSampler
 		// but replacing deciding based on traceId with sessionId
 		// (not extended from due to private methods)
-		const currentSession = getRumSessionId({ useLocalStorage: this.useLocalStorageForSessionMetadata })
-		if (this._currentSession !== currentSession) {
-			this._currentSessionSampled = this._accumulate(currentSession) < this._upperBound
-			this._currentSession = currentSession
+
+		// TODO: we are guaranteed to have a session here, so maybe error explicitly if that's not the case?
+
+		const currentSessionId = getOrInitShadowSession().id;
```

Comment:
First we create a shadow session, which we could also call "inactive". Wherever we consider if a session is active (or exists), this `shadow` attribute is checked first.